content/influxdb/clustered/write-data/best-practices/data-lifecycle.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Data ingest lifecycle best practices
+seotitle: InfluxDB best practices for data ingest lifecycle management
+description: >
+  Best practices for managing the lifecycle of data ingested into InfluxDB.
+menu:
+  influxdb_clustered:
+    name: Data ingest lifecycle
+    weight: 201
+    parent: write-best-practices
+---
+
+Data ingested into InfluxDB must conform to the retention period set on the database.
+Any data which lies outside of this retention period is no longer considered for retrieval by a query, but can have references maintained in object storage or the Postgres catalog, causing an increase in operational overhead and drive higher costs.
+
+In order to reduce these factors, it is important to manage the lifecycle of ingested data. You can achieve this by following the best practices below:
+
+- Use a retention policy which is appropriate for your requirements.
+- Tune the garbage collector cutoff period to ensure that data is removed in a timely manner.
+- Enable versioning in your object store and use lifecycle rules to expire old data.
```

Comment:
> no need to enable object versioning to manage the file lifecycle

I was wondering whether this was being conflated as I was writing it.

My understanding was that the object versioning _enables_ our lower value of `6h` in Serverless/Dedicated, but perhaps that actually isn't quite right? Versioning is merely a mechanism for GC to delete things and creates a new "version" on the deletion - the catalog can be smaller because of this though.

The deleted version being around is a helper for DR, although from speaking with Raphael a bit, and as you say, the tooling is a little lacking for restores.

Is that a better/more correct understanding?

Reply:
> My understanding was that the object versioning enables our lower value of 6h in Serverless/Dedicated, but perhaps that actually isn't quite right?

Note quite - lets assume that you, as a customer, do not care about backups - there's no point keeping any file longer than you need it. As soon as a file is no longer queryable (expired due to being outside the retention period) then there's no point in keeping that file. It can be deleted immediately![^kinda] Set your GC retention to 6h and don't bother with bucket versioning - you're never gonna read the thing.

You only need to keep old files around, if you need to access them again in the future - you only need to do that if you roll back the catalog to an older version ("restoring a backup"). Keeping old files is **only for backup purposes**.

If you do want backups, then you need those old files in case you roll back! It becomes a question of how and when those files get deleted - either delete after 30days, or delete them now and let S3 delete them in 30 days. In either case, you only need them for backup purposes!

[^kinda]: Ignoring outstanding queries and caches, so not quite immediately, but close enough.